topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -1195,21 +1206,20 @@ private void updateStatusForException(ReconcilableTopic reconcilableTopic, Excep
     }
 
     private void updateStatus(ReconcilableTopic reconcilableTopic) {
-        if (!TopicOperatorUtil.isManaged(reconcilableTopic.kt())) {
-            return;
-        }
         var oldStatus = Crds.topicOperation(kubeClient)
             .inNamespace(reconcilableTopic.kt().getMetadata().getNamespace())
             .withName(reconcilableTopic.kt().getMetadata().getName()).get().getStatus();
-        if (statusChanged(reconcilableTopic.kt(), oldStatus)) {
+
+        StatusDiff statusDiff = new StatusDiff(oldStatus, reconcilableTopic.kt().getStatus());
+        if (!statusDiff.isEmpty() || differentConditions(oldStatus.getConditions(), reconcilableTopic.kt().getStatus().getConditions())) {
```

Comment:
The StatusDiff should include the conditions. So why do we need the second part of the condition?

Reply:
You are right, we can remove it, I misslooked how StatusDiff evaluates the conditions path.

I'll remove it and have a look at the tests as well.